os/arch/arm/src/imxrt/imxrt_hyperflash.c

```diff
@@ -0,0 +1,673 @@
+													Speed Field: max(200MHz)
+													Open Drain Enable Field: Open Drain Disabled
+													Pull / Keep Enable Field: Pull/Keeper Enabled
+													Pull / Keep Select Field: Keeper
+													Pull Up / Down Config. Field: 100K Ohm Pull Down
+													Hyst. Enable Field: Hysteresis Disabled */
+	imxrt_iomuxc_setpinconfig(
+		IOMUXC_GPIO_SD_B1_09_FLEXSPIA_DATA01,   /* GPIO_SD_B1_09 PAD functional properties : */
+		0x10F1u);                               /* Slew Rate Field: Fast Slew Rate
+													Drive Strength Field: R0/6
+													Speed Field: max(200MHz)
+													Open Drain Enable Field: Open Drain Disabled
+													Pull / Keep Enable Field: Pull/Keeper Enabled
+													Pull / Keep Select Field: Keeper
+													Pull Up / Down Config. Field: 100K Ohm Pull Down
+													Hyst. Enable Field: Hysteresis Disabled */
+	imxrt_iomuxc_setpinconfig(
+		IOMUXC_GPIO_SD_B1_10_FLEXSPIA_DATA02,   /* GPIO_SD_B1_10 PAD functional properties : */
+		0x10F1u);                               /* Slew Rate Field: Fast Slew Rate
+													Drive Strength Field: R0/6
+													Speed Field: max(200MHz)
+													Open Drain Enable Field: Open Drain Disabled
+													Pull / Keep Enable Field: Pull/Keeper Enabled
+													Pull / Keep Select Field: Keeper
+													Pull Up / Down Config. Field: 100K Ohm Pull Down
+													Hyst. Enable Field: Hysteresis Disabled */
+	imxrt_iomuxc_setpinconfig(
+		IOMUXC_GPIO_SD_B1_11_FLEXSPIA_DATA03,   /* GPIO_SD_B1_11 PAD functional properties : */
+		0x10F1u);                               /* Slew Rate Field: Fast Slew Rate
+													Drive Strength Field: R0/6
+													Speed Field: max(200MHz)
+													Open Drain Enable Field: Open Drain Disabled
+													Pull / Keep Enable Field: Pull/Keeper Enabled
+													Pull / Keep Select Field: Keeper
+													Pull Up / Down Config. Field: 100K Ohm Pull Down
+													Hyst. Enable Field: Hysteresis Disabled */
+}
+
+static inline void flexspi_clock_init(void)
+{
+    const clock_usb_pll_config_t g_ccmConfigUsbPll = {.loopDivider = 0U};
+
+    imxrt_clock_initusb1pll(&g_ccmConfigUsbPll);
+    imxrt_clock_initusb1pfd(kCLOCK_Pfd0, 26);   /* Set PLL3 PFD0 clock 332MHZ. */
+    imxrt_clock_setmux(kCLOCK_FlexspiMux, 0x3); /* Choose PLL3 PFD0 clock as flexspi source clock. */
+    imxrt_clock_setdiv(kCLOCK_FlexspiDiv, 3);   /* flexspi clock 83M, DDR mode, internal clock 42M. */
+}
+
+/************************************************************************************
+ * Public Functions
+ ************************************************************************************/
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+/* The smallest integer type that can hold the IOMUX encoding */
+
+typedef uint16_t iomux_pinset_t;
+#ifdef CONFIG_MTD_PROGMEM
```

Comment:
Your code means board should use only this flash driver thorugh the progmem.
If you can't support any other flash(c.f. m25p), it can be acceptable, otherwise please implement specific flash driver to support mtd based on spi.

Reply:
In case of iMXRT Series, FlexSPI interface is currently used for flash driver.
FlexSPI is flexible between NOR flash and Hyper flash.
Now, we are using Hyper flash in evk.
If customer wants to use a kind of NOR flash, we will change the configuration by referring to CONFIG_EVK_NOR_FLASH
I want to keep the current status.